src/OrleansAWSUtils/Reminders/DynamoDBReminderTable.cs

```diff
@@ -0,0 +1,294 @@
+            }
+        }
+
+        public async Task TestOnlyClearTable()
+        {
+            var expressionValues = new Dictionary<string, AttributeValue>
+                {
+                    { $":{SERVICE_ID_PROPERTY_NAME}", new AttributeValue(serviceId.ToString()) }
+                };
+
+            try
+            {
+                var expression = $"{SERVICE_ID_PROPERTY_NAME} = :{SERVICE_ID_PROPERTY_NAME}";
+                var records = await storage.ScanAsync(TABLE_NAME_DEFAULT_VALUE, expressionValues, expression,
+                    item => new Dictionary<string, AttributeValue>
+                    {
+                        { REMINDER_ID_PROPERTY_NAME, item[REMINDER_ID_PROPERTY_NAME] },
+                        { GRAIN_HASH_PROPERTY_NAME, item[GRAIN_HASH_PROPERTY_NAME] }
+                    }).ConfigureAwait(false);
+
+                if (records.Count <= 25)
+                {
+                    await storage.DeleteEntriesAsync(TABLE_NAME_DEFAULT_VALUE, records);
+                }
+                else
+                {
+                    List<Task> tasks = new List<Task>();
+                    foreach (var batch in records.BatchIEnumerable(25))
+                    {
+                        tasks.Add(storage.DeleteEntriesAsync(TABLE_NAME_DEFAULT_VALUE, batch));
+                    }
+                    await Task.WhenAll(tasks);
+                }
+            }
+            catch (Exception exc)
+            {
+                logger.Warn(ErrorCode.ReminderServiceBase,
+                    $"Intermediate error removing reminder entries {Utils.DictionaryToString(expressionValues)} from table {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<string> UpsertRow(ReminderEntry entry)
+        {
+            var reminderId = ConstructReminderId(serviceId, entry.GrainRef, entry.ReminderName);
+            var keys = new Dictionary<string, AttributeValue>
+                {
+                    { REMINDER_ID_PROPERTY_NAME, new AttributeValue(reminderId) },
+                    { GRAIN_HASH_PROPERTY_NAME, new AttributeValue { N = entry.GrainRef.GetUniformHashCode().ToString() } }
+                };
+
+            var fields = new Dictionary<string, AttributeValue>
+                {
+                    { SERVICE_ID_PROPERTY_NAME, new AttributeValue(serviceId.ToString()) },
+                    { GRAIN_REFERENCE_PROPERTY_NAME, new AttributeValue( entry.GrainRef.ToKeyString()) },
+                    { PERIOD_PROPERTY_NAME, new AttributeValue(entry.Period.ToString()) },
+                    { START_TIME_PROPERTY_NAME, new AttributeValue(entry.StartAt.ToString()) },
+                    { REMINDER_NAME_PROPERTY_NAME, new AttributeValue(entry.ReminderName) }
+                };
+
+            var extraExpression = $"{ETAG_PROPERTY_NAME} = {ETAG_PROPERTY_NAME} + {ETAG_INCREMENT}";
+            var etagField = new Dictionary<string, AttributeValue>
```

Comment:
AFAIK, the reminder upsert in orleans should be unconditional and return the new unique etag, this is what's being done here. Only on the delete operation we check the etag. It never made sense to me, I've opened an issue about it a long time ago.